common/dal2/src/ops/io.rs

```diff
@@ -37,3 +99,24 @@ impl futures::AsyncRead for CallbackReader {
         r
     }
 }
+
+#[cfg(test)]
+mod test {
+    use futures::io::Cursor;
+    use futures::StreamExt;
+
+    use super::*;
+
+    #[tokio::test]
```

Comment:
This test should move to test :)

Reply:
`ReaderStream` intends to be used internally in `dal2` crate.

Do you think it's better to export it and then do integration testing?